calico/felix/fetcd.py

```diff
@@ -129,6 +149,67 @@ def _periodically_resync(self):
             gevent.sleep(sleep_time)
             self.force_resync(reason="periodic resync", async=True)
 
+    @logging_exceptions
+    def _status_reporting(self):
+        """
+        Greenlet: periodically writes to etcd
+
+        :return: Does not return, unless reporting disabled.
+        """
+        _log.info("Started status reporting thread.")
+        ttl = self._config.REPORTING_TTL_SECS
+        interval = self._config.REPORTING_INTERVAL_SECS
+        _log.info("Config loaded, reporting interval: %s, TTL: %s", interval, ttl)
+        if interval == 0:
+            _log.info("Interval is 0, status reporting disabled.")
+            return
+        while True:
+            try:
+                self.update_felix_status(ttl)
+                _log.info("Status updated.")
+                gevent.sleep(interval)
+            except EtcdException as e:
+                _log.info("Status updating failed. %r. Reconnecting... ", e)
+                _reconnect(self)
+                gevent.sleep(RETRY_DELAY)
+
+
+    def update_felix_status(self, ttl):
+        """
+        Writes status info to etcd
```

Comment:
Should explain that we're writing two things to etcd - uptime and a json doc containing the status info (which is currently just the current time).